plugins/module_utils/_vmware.py

```diff
@@ -368,88 +374,94 @@ def get_folder_by_absolute_path(self, folder_path, fail_on_missing=False):
                 recurse: If true, the search will recurse through the folder structure
             Returns:
-                dicttionary of {obj: str}. The keys are the object while the values are the
-                object name
+                list of objs
         """
         if not folder:
             folder = self.content.rootFolder
 
-        obj = {}
+        objs = []
         container = self.content.viewManager.CreateContainerView(folder, vimtype, recurse)
         for managed_object_ref in container.view:
             try:
-                obj.update({managed_object_ref: managed_object_ref.name})
+                objs += [managed_object_ref]
             except vmodl.fault.ManagedObjectNotFound:
                 pass
-        return obj
+        return objs
 
     def get_all_vms(self, folder=None, recurse=True):
         """
-        Get all virtual machines.
+            Get all virtual machines in a folder. Can recurse through folder tree if needed. If no folder
+            is provided, then the datacenter root folder is used
+            Args:
+                folder: vim.Folder, the folder object to use as a base for the search. If
+                        none is provided, the datacenter root will be used
+                recurse: If true, the search will recurse through the folder structure
+            Returns:
+                list of vim.VirtualMachine
         """
-        return self.list_all_objs_by_type([vim.VirtualMachine], folder=folder, recurse=recurse)
+        return self.get_all_objs_by_type([vim.VirtualMachine], folder=folder, recurse=recurse)
 
-    def get_datacenter_by_name(self, dc_name, fail_on_missing=False):
+    def get_datacenter_by_name_or_moid(self, identifier, fail_on_missing=False):
         """
             Get the datacenter matching the given name. Datacenter names must be unique
             in a given vcenter, so only one object is returned at most.
             Args:
-                dc_name: Name of the datacenter to search for
+                identifier: Name or MOID of the datacenter to search for
                 fail_on_missing: If true, an error will be thrown if no datacenters are found
             Returns:
                 datacenter object or None
         """
-        ds = self.get_objs_by_name_or_moid([vim.Datacenter], dc_name)
+        ds = self.get_objs_by_name_or_moid([vim.Datacenter], identifier)
         if not ds and fail_on_missing:
-            self.module.fail_json("Unable to find datacenter with name %s" % dc_name)
+            self.module.fail_json("Unable to find datacenter with name or MOID %s" % identifier)
         return ds
 
-    def get_cluster_by_name(self, cluster_name, fail_on_missing=False, datacenter=None):
+    def get_cluster_by_name_or_moid(self, identifier, fail_on_missing=False, datacenter=None):
         """
             Get the cluster matching the given name. Cluster names must be unique
```

Comment:
```suggestion
            Get the cluster matching the given name or moid. Cluster names must be unique
```

Reply:
this was missing in a few other places, i added it there as well